src/app/core/components/redeem-gift/redeem-gift.component.ts

```diff
@@ -0,0 +1,91 @@
+/* @format */
+import { Component, Input, OnInit } from '@angular/core';
+import {
+  UntypedFormGroup,
+  UntypedFormBuilder,
+  Validators,
+} from '@angular/forms';
+import { PromoVOData } from '@models';
+import { ApiService } from '@shared/services/api/api.service';
+import {
+  BillingResponse,
+  AccountResponse,
+} from '@shared/services/api/index.repo';
+import { FileSizePipe } from '@shared/pipes/filesize.pipe';
+import { AccountService } from '@shared/services/account/account.service';
+import { EventService } from '@shared/services/event/event.service';
+
+interface StorageRedemptionMessage {
+  successful: boolean;
+  message: string;
+}
+
+@Component({
+  selector: 'pr-redeem-gift',
+  templateUrl: './redeem-gift.component.html',
+  styleUrl: './redeem-gift.component.scss',
+})
+export class RedeemGiftComponent implements OnInit {
+  @Input() public promoCode: string = '';
+  public resultMessage: StorageRedemptionMessage | undefined;
+  public promoForm: UntypedFormGroup;
+  public waiting: boolean;
+
+  constructor(
+    private fb: UntypedFormBuilder,
```

Comment:
I know this is a pre-existing variable name, but this is probably a good opportunity to change it to something more informative.